### ISSUE TITLE
toDate() method with "." literal returns never

### ISSUE THREAD

**Issue by tomferez** [benwest/kql-ts#2]
Hi,

When using the toDate() method with a IntlDateFormatter that uses a "." on a field level, the returned type is never.

Example: `date: "page.date.toDate('d.m.Y')"` returns "never" - iso "string | number".

Am no expert of Typescript so couldn't really help with the issue...

Thanks!
Tom

**Comment by benwest** [comment]
Hi! Cool that you’re using the library 😃

You’re right that the dots are causing the problem. I think you should be able to write `date: "page.date.toDate('d.m.Y')" as "page.date.toDate"` to get around it.

**Comment by tomferez** [comment]
Yes I'm using it because the lack of "native" Typescript support is really what's preventing me from using KirbyCMS in a prod level with headless.

> I think you should be able to write `date: "page.date.toDate('d.m.Y')" as "page.date.toDate"` to get around it.

I thought of that, but then you lose the ability to format the date through the query. No big deal though... ☺️

**Comment by benwest** [comment]
> I thought of that, but then you lose the ability to format the date through the query. No big deal though... ☺️

no!- you're sending the same query, the `as` is a type hint.

But for what it's worth I've gone back to using zod myself to validate types because of this kind of thing :)
